src/Yubikey/Validate.php

```diff
@@ -269,7 +269,9 @@ public function check($otp, $multi = false)
 
         $clientId = $this->getClientId();
         if ($clientId === null) {
-            throw new \InvalidArgumentException('Client ID cannot be null');
+            if (!$this->getUseSecure())
+                throw new \InvalidArgumentException('Client ID cannot be null');
+            $clientId = 1;
```

Comment:
But the clientId is required to do the request. If you want put more
beautiful, you could put random integer (0-9999).
https://demo.yubico.com/php-yubico/demo.php

On 18/03/16 12:32, Chris Cornutt wrote:
>
> In src/Yubikey/Validate.php
> <https://github.com/enygma/yubikey/pull/20#discussion_r56643547>:
>
> > @@ -269,7 +269,9 @@ public function check($otp, $multi = false)
> >  
> >          $clientId = $this->getClientId();
> >          if ($clientId === null) {
> > -            throw new \InvalidArgumentException('Client ID cannot be null');
> > +            if (!$this->getUseSecure())
> > +                throw new \InvalidArgumentException('Client ID cannot be null');
> > +            $clientId = 1;
>
> I think refactoring things a bit here to just not add the |clientId|
> value to the |$params| array might be a better way to handle this than
> hard-coding an ID like this.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/enygma/yubikey/pull/20/files/67816d0b4177845c29a2370c7aa62fc3fbc95997#r56643547>
>